Mail/Hailgun/Message.hs

```diff
@@ -46,7 +46,8 @@ hailgunMessageWithReplyTo subject content sender recipients simpleAttachments re
    attachments <- attachmentsInferredFromMessage content cleanAttachments
    replyToAddr <- validateRecipient `traverse` replyTo
    return HailgunMessage
-      { messageSubject = subject
+      { messageId = Nothing
```

Comment:
It looks like this API encourages doing a record update on the result produced by `hailgunMessageWithReplyTo` to set the `messageId` to a value. This is all cool, but food for thought on if we want to make this an explicit, up-front choice for the user, like a `hailgunMessageWith :: MessageOpts -> Either HailgunErrorMessage HailgunMessage`, where `MessageOpts` would capture this function's params and the new message ID.

On the other hand, considering we're in a bit of a fork hole with this lib, it may be intentional/desirable for the message ID to be set after the fact in an effort to keep the fork's existing API as close to upstream as possible.

Reply:
Yeah I think we should probably create a new interface for `hailgunMessage` that takes a record of fields vs requiring the record update. Its not clear who really "owns" this anymore. Mercury might be the best bet and we can try opening an issue on their repo for a discussion if they want to assume maintenance and we then submit some updates.